crates/fiber-lib/src/store/store_impl/mod.rs

```diff
@@ -602,6 +604,21 @@ impl PreimageStore for Store {
         batch.commit();
     }
 
+    #[cfg(feature = "watchtower")]
+    fn get_preimage(&self, payment_hash: &Hash256) -> Option<Hash256> {
+        let key = [&[PREIMAGE_PREFIX], payment_hash.as_ref()].concat();
+        self.get(key)
+            .map(|v| deserialize_from(v.as_ref(), "Preimage"))
+            // Try to get the preimage from watchtower store
+            .or_else(|| {
+                let prefix = [&[WATCHTOWER_PREIMAGE_PREFIX], payment_hash.as_ref()].concat();
+                let mut iter = self.prefix_iterator(prefix.as_slice());
+                iter.next()
+                    .map(|(_key, value)| deserialize_from(value.as_ref(), "Preimage"))
```

Comment:
```suggestion
                    .map(|(_key, value)| deserialize_from(value.as_ref(), "Watchtower Preimage"))
```
to be different with error messages coming above.